File: DomainSpecificLanguages/DomainSpecificLanguages/AlarmsNestedFunctions/Outbound/XmlOutboundMessageFactory.cs
namespace TelecomSystemNestedFunctions.Outbound
{
  public interface IOutboundMessageFactory
  {
    XmlOutboundMessage CreateOutboundMessage();
  }

  public class XmlOutboundMessageFactory : IOutboundMessageFactory
  {
    public XmlOutboundMessage CreateOutboundMessage()
    {
      return new XmlOutboundMessage(new XmlMarshalling());
    }
  }
}

File: Java/dependency-injection/_03_control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/Outbound/XmlOutboundMessageFactory.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.Outbound;

public class XmlOutboundMessageFactory implements OutboundMessageFactory {
  public OutboundMessage createOutboundMessage() {
    return new XmlOutboundMessage(new XmlMarshalling());
  }
}
